lib/qs/qs_runner.rb

```diff
@@ -17,6 +32,18 @@ def run_callbacks(callbacks)
       callbacks.each{ |proc| self.handler.instance_eval(&proc) }
     end
 
+    module OptionalTimeout
+      def self.after(timeout, &block)
```

Comment:
@jcredding open to other suggestions as well.  I also think just having a private method called `optionally_timeout_after` works too.  Something about that `.after` trips me up...